x/valset-pref/validator_set.go

```diff
@@ -156,6 +194,12 @@ func (k Keeper) PreformRedelegation(ctx sdk.Context, delegator sdk.AccAddress, e
 
 		tokenFromShares := validator.TokensFromShares(delegation.Shares)
 		existing_val, existing_val_zero_amount := k.GetValSetStruct(existingVals, tokenFromShares)
+		// Matt: Do we need to initiazlize them using this way in the first place?
+		// Here we're creating a copy of the val set struct with zero amount, then adding them to the other struct
+		// to compare the diffs.
+		// Is it not possible to not have this logic in the first place and then perform
+		// diffAmount := newVals.amount.Sub(newValSet[i].amount) right away?
+		// I feel like by using maps this problem can be solvable.
```

Comment:
the reason why i didn't use maps here is because we allow duplicate valAddrs as well which maps doesnot allow. For instance

if delegatorA with valSet [valA: 50osmo] wants to redelegate --> [valA: 20osmo, valB: 10osmo, valC: 20osmo]